src/common/xio_observer.c

```diff
@@ -198,7 +198,7 @@ void xio_observable_notify_observer(struct xio_observable *observable,
 				    struct xio_observer *observer,
 				    int event, void *event_data)
 {
-	if (likely(observable->impl && observer->impl))
+	if (likely(observable->impl && observer->impl && observer->notify))
```

Comment:
usually it happen when fin ack arrives after connection timeout occur. observers already freed.
so message  does not have observer and we may free this message.